### DISCUSSION TITLE
Wrong Boolean value (understanding numbers as true)

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#33833] [Q&A]
Hi, Can you answer me please when i added field Boolean in my class (quarkus project). For Example: 
public class MyClass {
    @NotBlank
    private String newVersion;
    @NotNull
    private AppName newName;
    @NotNull
    private Boolean isStart;
} 
 And when i want use it in my json request body

{
	"newVersion": "8.67.79.9",
	"newName": "NAME",
	"isStart": 45678
}

) and give initial value except true or false (example: 45678), then it understands numbers as true. How to fix the problem?

**Comment by Angel0926** [comment]
@MikeEdgar

**Comment by MikeEdgar** [comment]
If you're using Jackson, it's probably being coerced from integer to boolean. Zero == false, else true [1].

[1] https://github.com/joansmith/jackson-databind/blob/230a19d527df7dc0c05017429bcb7aa1cbbe41ef/src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java#L186-L193

**Reply by Angel0926** [reply]
no i want that it work only with true or false and in other occasion throw exception
